portfolio/src/main/java/com/google/sps/servlets/TranslateServlet.java

```diff
@@ -0,0 +1,64 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.cloud.translate.Translate;
+import com.google.cloud.translate.TranslateOptions;
+import com.google.cloud.translate.Translation;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servelet that translates inputted text */
+@WebServlet("/translate")
+public class TranslateServlet extends HttpServlet {
+
+    private static final String REQUEST_MESSAGE_PARAM = "message";
+    private static final String REQUEST_LANGUAGE_CODE_PARAM = "languageCode";
+    private static final String REQUEST_MOCK_PARAM = "mock";
+    private static final String RESPONSE_TEXT_CONTENT = "text/html; charset=UTF-8";
+    private static final String RESPONSE_CHAR_ENCODING = "UTF-8";
+
+    /**
+     * Gets the translated message based on front end text and language code
+     * @param request The request object
+     * @param response The response object
+     */
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Get the request parameters
+        String originalText = request.getParameter(REQUEST_MESSAGE_PARAM);
+        String languageCode = request.getParameter(REQUEST_LANGUAGE_CODE_PARAM);
```

Comment:
any chance these two would also be null?

Reply:
The one case that I discovered such that the originalText and/or languageCode could be null is when I edit the front-end JS code to alter the translateComments method. To prevent the uncaught nulls, I just did a check in the back-end to ensure both  request parameters are not null before proceeding. I also added a check to ensure that the language code is a valid code accepted by Translate, because I realized that the submitted language code could be edited by Inspecting Elements.